### DISCUSSION TITLE
what thread library can parallel version of Pardiso solver use

### DISCUSSION THREAD

**Discussion by cjl88** [coin-or/Ipopt#461] [Q&A] [answered]
It looks like parallel version of Pardiso solver can only use OpenMP from either Intel (libiomp5) or GNU  (libgomp). Right?

Intel MKL says it can use TBB threading from Intel. But this is not used by parallel version of Pardiso solver.

I am referring specifically to this Pardiso library:
"If you want to compile Ipopt with the Pardiso library that is included in Intel MKL, it is sufficient to ensure that MKL is used for the linear algebra routines (Blas/Lapack), see BLAS and LAPACK."

Thanks.

**Comment by svigerske** [comment, marked as answer]
You might want to check this with the Intel MKL project. I don't know much about what threading libraries a third-party linear solver is supporting. OpenMP worked for me.

**Reply by cjl88** [reply]
> You might want to check this with the Intel MKL project. I don't know much about what threading libraries a third-party linear solver is supporting. OpenMP worked for me.

As I mentioned in another post, I am having solver convergence issues with OpenMP from both Intel and GNU. So I am looking at ways other than OpenMP.

I set env var OMP_NUMTHREADS=4. That's the only thing I did to enable parallel Pardiso.

Thanks.
